crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes.rs

```diff
@@ -0,0 +1,1373 @@
+use std::vec;
+
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, AddVisitor, AnalyzerOptions, FixKind,
+    Phases, QueryMatch, Queryable, Rule, RuleDiagnostic, RuleKey, ServiceBag, Visitor,
+    VisitorContext,
+};
+use biome_console::markup;
+use biome_deserialize::{
+    Deserializable, DeserializableValue, DeserializationDiagnostic, DeserializationVisitor, Text,
+    VisitableType,
+};
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{js_string_literal, js_string_literal_expression, jsx_string};
+use biome_js_syntax::{
+    JsCallArguments, JsCallExpression, JsLanguage, JsStringLiteralExpression,
+    JsTemplateChunkElement, JsxAttribute, JsxString,
+};
+use biome_rowan::{AstNode, BatchMutationExt, Language, SyntaxNode, TextRange, WalkEvent};
+use rustc_hash::FxHashMap;
+
+use crate::JsRuleAction;
+
+// TODO: variants
+// TODO: extensibility
+// TODO: preset generation script
+// TODO: automatic config sync
+// TODO: remove duplicated classes
+
+const CLASS_ATTRIBUTES: [&str; 2] = ["class", "className"];
+
+// rule metadata
+// -------------
+
+declare_rule! {
+    /// Enforce the sorting of CSS classes.
+    ///
+    /// TODO: description
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div class="px-2 foo px-4 bar" />;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// // TODO: examples
+    /// ```
+    ///
+    pub(crate) UseSortedClasses {
+        version: "next",
+        name: "useSortedClasses",
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+// utils
+// -----
+
+fn get_callee_name(call_expression: &JsCallExpression) -> Option<String> {
+    Some(
+        call_expression
+            .callee()
+            .ok()?
+            .as_js_identifier_expression()?
+            .name()
+            .ok()?
+            .name()
+            .ok()?
+            .to_string(),
+    )
+}
+
+fn get_attribute_name(attribute: &JsxAttribute) -> Option<String> {
+    Some(attribute.name().ok()?.as_jsx_name()?.to_string())
+}
+
+fn is_call_expression_of_valid_function(
+    call_expression: &JsCallExpression,
+    functions: &[String],
+) -> bool {
+    match get_callee_name(call_expression) {
+        Some(name) => functions.contains(&name.to_string()),
+        None => false,
+    }
+}
+
+fn is_valid_attribute(attribute: &JsxAttribute, attributes: &[String]) -> bool {
+    match get_attribute_name(attribute) {
+        Some(name) => attributes.contains(&name.to_string()),
+        None => false,
+    }
+}
+
+fn get_options_from_analyzer(analyzer_options: &AnalyzerOptions) -> UseSortedClassesOptions {
+    match analyzer_options
+        .configuration
+        .rules
+        .get_rule_options::<UseSortedClassesOptions>(&RuleKey::new("nursery", "useSortedClasses"))
+    {
+        Some(options) => options.clone(),
+        None => UseSortedClassesOptions::default(),
+    }
+}
```

Comment:
So, the visitor needs to know about the options because you can specify custom functions or attributes to match class strings. You can read about the equivalent options in the prettier plugin here: https://github.com/tailwindlabs/prettier-plugin-tailwindcss#sorting-classes-in-function-calls

In the visitor, the context gives you the options for the whole thing, so I have to access this rule's options through the `get_rule_options` method.